lib/db/migrator/migrator.go

```diff
@@ -33,26 +37,42 @@ func MigrateWithMigrationsTable(pgURL, migrationsPath, migrationsTable string, l
 	if err != nil {
 		return errors.Wrap(err, "parse PG URL")
 	}
+
 	m, err := migrate.New(addScheme(migrationsPath), purl)
 	if err != nil {
 		return errors.Wrap(err, "init migrator")
 	}
 	m.Log = migrationLog{Logger: l, verbose: verbose}
+
+	// Plan 1:
+	// get a database driver from the migration package and wrap the version checky stuff in a lock
+	databaseDrv, err := database.Open(purl)
```

Comment:
Here we're getting a second database driver, different from the one owned by the `m` struct. We could instead create the `m` migrator with the `migrate.NewWithDatabaseInstance` function https://github.com/golang-migrate/migrate/blob/31c22870d306d0facce2b70c291b458c007b327e/migrate.go#L121 But that would not work with the code as written, as the driver won't let you Lock if you are already locked. https://github.com/golang-migrate/migrate/blob/31c22870d306d0facce2b70c291b458c007b327e/database/postgres/postgres.go#L157-L176